azure.go

```diff
@@ -324,20 +324,24 @@ func (ac *AzureClient) listFromResourceGroup(resourceGroup string, resourceTypes
 }
 
 // Returns all resource with the given couple tagname, tagvalue
-func (ac *AzureClient) listByTag(tagName string, tagValue string, types []string) ([]AzureResource, error) {
+func (ac *AzureClient) listByTag(tagName string, tagValue string, types []string, resourcesMap map[string][]byte) ([]AzureResource, error) {
 	apiVersion := "2018-05-01"
 	securedTagName := secureString(tagName)
 	securedTagValue := secureString(tagValue)
 	filterTypes := url.QueryEscape(fmt.Sprintf("tagName eq '%s' and tagValue eq '%s'", securedTagName, securedTagValue))
 	subscription := fmt.Sprintf("subscriptions/%s", sc.C.Credentials.SubscriptionID)
 	resourcesEndpoint := fmt.Sprintf("%s/%s/resources?api-version=%s&$filter=%s", sc.C.ResourceManagerURL, subscription, apiVersion, filterTypes)
 
-	body, err := getAzureMonitorResponse(resourcesEndpoint)
-	if err != nil {
-		return nil, err
+	body, dataExists := resourcesMap[resourcesEndpoint]
+	data := AzureResourceListResponse{}
+	var err error
+	if !dataExists {
+		body, err = getAzureMonitorResponse(resourcesEndpoint)
+		if err != nil {
+			return nil, err
+		}
+		resourcesMap[resourcesEndpoint] = body
```

Comment:
Have you consider caching the result of `data.extendResources()` instead of the raw body (so that caching also save `Unmarshal` process, and `extendResources()` process ?

Reply:
Cannot do that. You can pass different types to that request and the only thing I can effectively cache is the result of the request.
So I tried to use the marshalled version of the body but I get pointer errors, strings that have not coherent.

Reply:
Ok!